test/test_jit.py

```diff
@@ -355,21 +355,30 @@ def f(x, y):
         self.assertExpectedGraph(trace)
         self.assertExportImport(trace, (x, y))
 
-    def test_peephole(self):
+    def test_typeas_trace_check(self):
         a = torch.tensor([0.4], requires_grad=True)
         b = torch.tensor([0.7], requires_grad=True)
+
+        def f(x, y):
+            return x.type_as(y)
+
+        trace = torch.jit.trace(f, (a, b))
```

Comment:
What's so special about this test?

Reply:
A cascade of things:
1. I updated the `test_peephole` tests to use torch.jit.trace because the other one doesn't DCE and was causing some dead copy_ code to show up (type_as harmlessly internally uses copy_).
2. This uncovered an assert failure when `requires_grad=True` on an older master. So I added this test to check for that. Updating to newest master fixes the problem, but it seems like a good idea to leave the test.